Source/WebKit/NetworkProcess/soup/WebSocketTaskSoup.cpp

```diff
@@ -108,6 +123,41 @@ WebSocketTask::~WebSocketTask()
     cancel();
 }
 
+void WebSocketTask::authenticate(WebCore::AuthenticationChallenge&& challenge)
```

Comment:
We'll need to find some way to avoid the code duplication between here and `NetworkDataTaskSoup::completeAuthentication`. Hmm....

Reply:
Yes, also thought that, but not sure if I have enough insight into the project to do so.

Reply:
Hm, it looks like the switch statement can be split off into a separate function that takes a `const Credential&` and a `SoupSession*` parameter. Then the only question is where to put the function. Let's put it in a new file under Source/WebKit/NetworkProcess/soup, which can contain just the one free function. As to what to call the file, let's call it SoupAuthentication.h and SoupAuthentication.cpp, unless you can think of a better name for it. It's not quite a perfect name because there's a lot of authentication code that's only in NetworkDataTaskSoup.cpp, but seems good enough unless anybody else has a better idea.

Since this new file is not cross-platform, you only have to add it to SourcesGTK.txt and SourcesWPE.txt and don't need to worry about modifying XCode build files. Should be easy... unless it triggers unified build failures due to shifting unified source bundles, but don't need to worry about that unless it happens.

BTW, I see NetworkDataTaskSoup has a bunch of code to cancel the authentication, but WebSocketTaskSoup does not. Is that because web sockets cannot be redirected?

Reply:
Will call it AuthenticationSoup rather, an update to the PR will follow.

Reply:
Implemented, but I can't fix this check:
```
ERROR: Source/WebKit/NetworkProcess/soup/AuthenticationSoup.cpp:48:  ASSERT_NOT_REACHED() statement fallthrough may result in unexpected code execution.  [security/assertion_fallthrough] [4]
ERROR: Source/WebKit/NetworkProcess/soup/AuthenticationSoup.cpp:58:  ASSERT_NOT_REACHED() statement fallthrough may result in unexpected code execution.  [security/assertion_fallthrough] [4]
ERROR: Source/WebKit/NetworkProcess/soup/AuthenticationSoup.cpp:83:  ASSERT_NOT_REACHED() statement fallthrough may result in unexpected code execution.  [security/assertion_fallthrough] [4]
ERROR: Source/WebKit/NetworkProcess/soup/AuthenticationSoup.cpp:90:  ASSERT_NOT_REACHED() statement fallthrough may result in unexpected code execution.  [security/assertion_fallthrough] [4]
```

Seems that the checker should have an ignore for these lines.

Reply:
Just dropped the ELSE case, the ASSERT_NOT_REACHED is anyway compiled to (void)0.

Reply:
That looks like a bug in the style checker? Seems to be clearly incorrect? Dropping a useful assert is unfortunate. Try changing it to `RELEASE_ASSERT_NOT_REACHED()` instead.

Reply:
Fixed the style check by adding inline methods.